### PR TITLE
Add local store of issue rule details with controller to view

### PR DESCRIPTION
**Pull request by reggieb**
Jira [GDX-731](https://cddodatamarketplace.atlassian.net/browse/GDX-731)

## Description
With this change in place the details links in the Spectral reports link to:
- the local rule path if the details are defined in the rules locale
- a remote site which lists standard spectral rule details

The definitions of what will appear on each page are defined in config/locale/rules.yml. The content can use Govspeak markdown to provide reach content. Currently that is just an h2 followed by a line of text.

## Production change
Note that a new environmental variable has been added `REMOTE_RULE_LIST_URL` This will need to be set as `"https://meta.stoplight.io/docs/spectral/4dec24461f3af-open-api-rules"` in the staging and production environments.

For example the details for an oas3-strict rule will link to this page:

rule_detail_page

[GDX-731]: https://cddodatamarketplace.atlassian.net/browse/GDX-731?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/helpers/rules_helper.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module RulesHelper
+  require "govspeak"
```
Comment (duckworthj39) [inline app/helpers/rules_helper.rb:4]: [suggestion]
I don't think you need this line, it should be required throughout the project

**Review by duckworthj39 — COMMENTED** [review]
Just some very minor comments

File: app/helpers/rules_helper.rb (hunk 1 above)
Comment (RobNicholsGDS) [reply app/helpers/rules_helper.rb:4]: You're right. In the past I've had to specify it but it doesn't look like I need to here.

Good spot - thank you.

File: spec/requests/rules_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe "Rules", type: :request do
+
```
[the remark this answers is not in the document]
Comment (RobNicholsGDS) [reply spec/requests/rules_spec.rb:4]: :+1:

File: spec/requests/rules_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      doc.to_html.html_safe
+    end
+
+    it "returns http success" do
+      get rule_path(rule)
+      expect(response).to have_http_status(:success)
+    end
+
+    it "outputs the matching translation as markdown" do
+      get rule_path('oas3-strict')
```
[the remark this answers is not in the document]
Comment (RobNicholsGDS) [reply spec/requests/rules_spec.rb]: :man_facepalming:

**Review by duckworthj39 — APPROVED** [review]
Looks good!

**Pull request merged**